diff --git a/src/main/resources/db/changelog/changesets/changelog_20241008T170740Z.xml b/src/main/resources/db/changelog/changesets/changelog_20241008T170740Z.xml
index 7180c5b9..a81987c1 100644
--- a/src/main/resources/db/changelog/changesets/changelog_20241008T170740Z.xml
+++ b/src/main/resources/db/changelog/changesets/changelog_20241008T170740Z.xml
@@ -137,11 +137,19 @@
 
     <!-- 7. migration Data -->
     <changeSet author="souissimai (generated)" id="1728312755079-30">
-                <sqlFile
-                        encoding="UTF-8"
-                        path="migrationDataTimePoint.sql"
-                        relativeToChangelogFile="true"
-                        splitStatements="true"
-                        stripComments="true"/>
+        <sqlFile
+                dbms="postgresql"
+                encoding="UTF-8"
+                path="migrationDataTimePointPostgres.sql"
+                relativeToChangelogFile="true"
+                splitStatements="true"
+                stripComments="true"/>
+        <sqlFile
+                dbms="!postgresql"
+                encoding="UTF-8"
+                path="migrationDataTimePoint.sql"
+                relativeToChangelogFile="true"
+                splitStatements="true"
+                stripComments="true"/>
     </changeSet>
 </databaseChangeLog>
diff --git a/src/main/resources/db/changelog/changesets/migrationDataTimePoint.sql b/src/main/resources/db/changelog/changesets/migrationDataTimePoint.sql
index f40b45ad..e909c4da 100644
--- a/src/main/resources/db/changelog/changesets/migrationDataTimePoint.sql
+++ b/src/main/resources/db/changelog/changesets/migrationDataTimePoint.sql
@@ -10,7 +10,7 @@ INSERT INTO time_point (
     study_uuid           -- UUID associated with the study (foreign key to study table)
 )
 SELECT
-    gen_random_uuid(),   -- Generate a new unique UUID for the time_point row
+    random_uuid(),   -- Generate a new unique UUID for the time_point row
     s.case_format,       -- Retrieve case_format from the study table
     s.case_name,         -- Retrieve case_name from the study table
     s.case_uuid,         -- Retrieve case_uuid from the study table
@@ -49,7 +49,7 @@ INSERT INTO time_point_node_info (
     voltage_init_result_uuid          -- UUID for voltage initialization result
 )
 SELECT
-    gen_random_uuid(),                -- Generate a new unique UUID for each time_point_node_info row
+    random_uuid(),                -- Generate a new unique UUID for each time_point_node_info row
     t.id,                             -- Reference the id of the associated time_point
     n.id_node,                        -- Reference the id of the associated network_modification_node_info
     n.dynamic_simulation_result_uuid, -- Retrieve dynamic simulation result UUID  from the table network_modification_node_info
